internals/utils/helper.go

```diff
@@ -62,6 +62,9 @@ func AppendUpperResourcePoolAnnotation(sns *ObjectContext, parentSns *ObjectCont
 		if err = sns.AppendAnnotations(map[string]string{danav1.IsUpperRp: danav1.True}); err != nil {
 			return err
 		}
+		if err = sns.DeleteAnnotations([]string{danav1.UpperRp}); err != nil {
+			return err
+		}
```

Comment:
I'm not sure we need to delete it? I think it would be fine a `upperRp` label to have this annotation, no?